File: frontend/src/api/map.ts
import api from './index'

export interface ProvinceMapStat {
  province: string
  total: number
  fake_count: number
  real_count: number
  unknown_count: number
  value: number
}

export interface ChinaMapDataResponse {
  provinces: ProvinceMapStat[]
  total_news: number
  total_fake: number
  updated_at: string
}

export interface ProvinceNewsItem {
  news_id: string
  title: string
  label: string
  platform: string
  publish_time: string
  location: string
}

export interface ProvinceDetailResponse {
  province: string
  stats: ProvinceMapStat
  items: ProvinceNewsItem[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

export interface ProvinceDetailParams {
  page?: number
  page_size?: number
  label?: string
}

export const mapApi = {
  async chinaGeoJson() {
    const { data } = await api.get('/map/china-geojson')
    return data
  },

  async chinaData() {
    const { data } = await api.get<ChinaMapDataResponse>('/map/china-data')
    return data
  },

  async provinceDetail(province: string, params: ProvinceDetailParams = {}) {
    const { data } = await api.get<ProvinceDetailResponse>(`/map/province/${encodeURIComponent(province)}`, { params })
    return data
  },
}

File: backend/app/schemas/map.py
from pydantic import BaseModel, Field


class ProvinceMapStat(BaseModel):
    province: str
    total: int = 0
    fake_count: int = 0
    real_count: int = 0
    unknown_count: int = 0
    value: int = 0


class ProvinceNewsItem(BaseModel):
    news_id: str
    title: str = ""
    label: str = ""
    platform: str = ""
    publish_time: str = ""
    location: str = ""


class ChinaMapDataResponse(BaseModel):
    provinces: list[ProvinceMapStat] = Field(default_factory=list)
    total_news: int = 0
    total_fake: int = 0
    updated_at: str = ""


class ProvinceDetailResponse(BaseModel):
    province: str
    stats: ProvinceMapStat
    items: list[ProvinceNewsItem] = Field(default_factory=list)
    total: int = 0
    page: int = 1
    page_size: int = 20
    total_pages: int = 0
